nncf/quantization/algorithms/weight_compression/torch_fx_backend.py

```diff
@@ -0,0 +1,239 @@
+from nncf.experimental.torch.fx.node_utils import get_graph_node_by_name
+from nncf.experimental.torch.fx.node_utils import get_tensor_constant_from_node
+from nncf.experimental.torch.fx.transformations import constant_update_transformation_builder
+from nncf.experimental.torch.fx.transformations import module_insertion_transformation_builder
+from nncf.parameters import CompressWeightsMode
+from nncf.quantization.algorithms.weight_compression.backend import WeightCompressionAlgoBackend
+from nncf.quantization.algorithms.weight_compression.config import WeightCompressionParameters
+from nncf.quantization.algorithms.weight_compression.lora_correction import LoraCorrectionAlgorithm
+from nncf.quantization.algorithms.weight_compression.torch_backend import PTWeightCompressionAlgoBackend
+from nncf.quantization.algorithms.weight_compression.weight_lowering import compress_weight
+from nncf.tensor import Tensor
+from nncf.tensor.definitions import TensorDataType
+from nncf.torch.graph import operator_metatypes as om
+from nncf.torch.graph.transformations.commands import PTTargetPoint
+from nncf.torch.model_graph_manager import find_const_node_in_constant_subgraph
+from nncf.torch.model_graph_manager import get_const_node
+from nncf.torch.quantization.layers import AsymmetricWeightsDecompressor
+from nncf.torch.quantization.layers import SymmetricWeightsDecompressor
+from nncf.torch.tensor_statistics.collectors import get_raw_stat_collector
+
+
+class FXWeightCompressionAlgoBackend(WeightCompressionAlgoBackend):
+    MATMUL_METATYPES = PTWeightCompressionAlgoBackend.MATMUL_METATYPES
+    EMBEDDING_METATYPES = PTWeightCompressionAlgoBackend.EMBEDDING_METATYPES
+    CONVOLUTION_METATYPES = PTWeightCompressionAlgoBackend.CONVOLUTION_METATYPES
+
+    @property
+    def matmul_metatypes(self) -> List[OperatorMetatype]:
+        return FXWeightCompressionAlgoBackend.MATMUL_METATYPES
+
+    @property
+    def embedding_metatypes(self) -> List[OperatorMetatype]:
+        return FXWeightCompressionAlgoBackend.EMBEDDING_METATYPES
+
+    @property
+    def convolution_metatypes(self) -> List[OperatorMetatype]:
+        return FXWeightCompressionAlgoBackend.CONVOLUTION_METATYPES
+
+    @staticmethod
+    def is_node_with_weights(node: NNCFNode, graph: NNCFGraph) -> bool:
+        return PTWeightCompressionAlgoBackend.is_node_with_weights(node, graph)
+
+    @staticmethod
+    def get_weight_names_and_port_ids(node: NNCFNode, graph: NNCFGraph) -> List[Tuple[str, int]]:
+        weight_port_ids = []
+        for prev_node in graph.get_previous_nodes(node):
+            weight_node = find_const_node_in_constant_subgraph(prev_node, graph)
+            if weight_node is None:
+                continue
+            edge = graph.get_edge(prev_node, node)
+            if edge.input_port_id in node.metatype.weight_port_ids:
+                weight_port_ids.append((weight_node.node_name, edge.input_port_id))
```

Comment:
Can we reuse this here?
https://github.com/openvinotoolkit/nncf/blob/develop/nncf/torch/model_graph_manager.py#L239-L253

Reply:
Maybe with some modification? but the function defined in `nncf/torch/model_graph_manager.py` returns only the list on port ids whereas both the node names and port ids are required here.

Reply:
Done!